versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/implrocksdb/RocksDBStoreManager.java

```diff
@@ -0,0 +1,278 @@
+        @Override
+        public RocksDBStore load(String name) throws RocksDBException {
+          return newStore(name);
+        }
+      });
+
+  public RocksDBStoreManager(String baseDirectory, boolean noDBOpenRetry) {
+    this.baseDirectory = baseDirectory;
+    this.noDBOpenRetry = noDBOpenRetry;
+  }
+
+  private RocksDBStore newStore(String name) throws RocksDBException {
+    final ColumnFamilyDescriptor columnFamilyDescriptor = new ColumnFamilyDescriptor(name.getBytes(UTF_8));
+    ColumnFamilyHandle handle = db.createColumnFamily(columnFamilyDescriptor);
+    handleIdToNameMap.put(handle.getID(), name);
+
+    return newRocksDBStore(name, columnFamilyDescriptor, handle);
+  }
+
+  private RocksDBStore newRocksDBStore(String name, ColumnFamilyDescriptor columnFamilyDescriptor,
+                                       ColumnFamilyHandle handle) {
+    /*
+    final RocksMetaManager rocksManager;
+    if (BLOB_WHITELIST.contains(name)) {
+      rocksManager = new RocksMetaManager(baseDirectory, name, FILTER_SIZE_IN_BYTES);
+    } else {
+      rocksManager = new RocksMetaManager(baseDirectory, name, Long.MAX_VALUE);
+    }
+    */
+    return new RocksDBStore(name, columnFamilyDescriptor, handle, db);
+  }
+
+  /**
+   * This method opens the RocksDB database. It creates a cache of the tables(Column Families) created.
+   */
+  @Override
+  public void start() {
+    //TODO need to make path configurable.
+    List<byte[]> families = null;
+
+    //TODO implement valiation of path existing
+
+    try (final Options options = new Options()) {
+      options.setCreateIfMissing(true);
+      // get a list of existing families.
+      families = new ArrayList<>(RocksDB.listColumnFamilies(options, baseDirectory));
+    } catch (RocksDBException e) {
+      LOGGER.error("Error extracting RocksDB options", e);
+    }
+
+    // if empty, add the default family
+    if (families != null && families.isEmpty()) {
+      families.addAll(COLUMN_FAMILIES);
+    }
+    final Function<byte[], ColumnFamilyDescriptor> func = ColumnFamilyDescriptor::new;
+
+    List<ColumnFamilyHandle> familyHandles = new ArrayList<>();
+    try (final DBOptions dboptions = new DBOptions()) {
+      dboptions.setCreateIfMissing(true);
+
+      // TODO what options need to be set?
+
+      // registerMetrics(dboptions);
+      try {
+        db = openDB(dboptions, baseDirectory, new ArrayList<>(Lists.transform(families, func)), familyHandles);
+
+      } catch (RocksDBException e) {
+        LOGGER.error("Error opening RocksDB backing store", e);
+      }
+    }
+    // populate the local cache with the existing tables.
+    for (int i = 0; i < families.size(); i++) {
+      byte[] family = families.get(i);
+      if (Arrays.equals(family, RocksDB.DEFAULT_COLUMN_FAMILY)) {
+        defaultHandle = familyHandles.get(i);
+      } else {
+        String name = new String(family, UTF_8);
+        final ColumnFamilyHandle handle = familyHandles.get(i);
```

Comment:
Where does familyHandles have values added to it?

Reply:
in start(), line 150. These are defined in COMUMN_FAMILIES.

Reply:
That adds to families, not familyHandles. familyHandles is created on line 154...